client/src/appsupport.res

```diff
@@ -0,0 +1,310 @@
+  // known platform
+  if (isMac) {
+    0
+  } else if (isLinux) {
+    1
+  } else if (isWindows) {
+    2
+  } else {
+    3
+  }
+}
+
+// ---------------------------
+// Rollbar
+// ---------------------------
+
+module Rollbar = {
+  type person ={
+    id: string,
+    username: string,
+  }
+
+  type rollbarConfig = {
+    enabled: bool,
+    payload: person,
+  }
+  type rollbar =
+  {
+  init: rollbarConfig => unit,
+  error: Js.Json.t => unit,
+  configure: rollbarConfig => unit,
+  }
+  @module external rollbar: rollbar = "rollbar"
+  let init = (rollbarConfig: rollbarConfig) => {
+    rollbar.init(rollbarConfig)
+  }
+  let error = (error: Js.Json.t) => {
+    rollbar.error(error)
+  }
+  let payload = (payload: person) => {
+    rollbar.configure({enabled: true, payload: payload})
+      }
+  let enabled = (enabled: bool) => {
+    rollbar.configure({enabled: enabled, payload: {id: "", username: ""}})
+  }
+  let configure = ({enabled: enabled, payload: payload}) => {
+    rollbar.configure({enabled: enabled, payload: payload})
+  }
+
+let username=""
+let userID=""
+  let rollbarConfig = {
+    enabled: true,
+    payload: { id: userID, username: username }
+  }
+let rollbarg = rollbar.init(rollbarConfig)
+
+let searchParams = Webapi.Url.URLSearchParams.make(window->Window.location->Location.href)
+if (searchParams->Webapi.Url.URLSearchParams.get("use-assets-tunnel") != None) {
+  rollbar.configure({...rollbarConfig, enabled: false})
+}
+
+
+let displayError = (msg)=>{
+  let event = CustomEvent.makeWithOptions(
+    "displayError",
+    {
+      "detail": {
+        "detail": msg,
+      },
+    },
+  )
+  document->Document.dispatchEvent(event)
+}
+
+}
+
+// ---------------------------
+// Pusher
+// ---------------------------
+
+
+module Pusher = {
+  type pusherConfig = {
+    cluster: string,
+    key: string,
+    forceTLS: bool,
+    enabled: bool,
+
+  }
+  type pusher = {
+    init: pusherConfig => unit,
+  }
+
+@module("pusher-js") external pusher: pusher = "Pusher"
+
+let pusherConfig = {
+  key: "",
+  cluster: "",
+  forceTLS: true,
+  enabled: true,
+}
+
+
+ if (pusherConfig.enabled) {
+  let pusherConnection = pusher.init(pusherConfig)
+  pusherConnection
+
+}
+}
+
+
+//window.Dark{}
+
+
+
+// ---------------------------
+// Focus
+// ---------------------------
+
+// mozhidden is not found in webapi
+
+let windowFocusChange = (visible) => {
+  let event = CustomEvent.makeWithOptions("windowFocusChange", { "detail": visible });
+  let target = Webapi.Dom.document |> Webapi.Dom.Document.asEventTarget
+  target->dispatchEvent(event);
+}
+
+let pageHidden = ref(false)
+
+let visibilityCheck=() => {
+  let hidden = ref(false)
+  if (Js.typeof(Webapi.Dom.document |> Document.hidden) !== "undefined") {
+    hidden := Webapi.Dom.document |> Document.hidden
+  }
+  // else if (typeof document.mozHidden !== "undefined") {
+  //   hidden = document.mozHidden
+  // } else if (typeof document.msHidden !== "undefined") {
```

Comment:
Couldn't find `document.mozHidden` in rescript-webapi.

Reply:
You can add them to webapi and use local versions of the definition here too.